pkg/envoy/ingress_translator.go

```diff
@@ -198,7 +198,12 @@ func translateIngresses(ingresses []v1beta1.Ingress) *envoyConfiguration {
 				}
 
 				envoyIngress := envoyIngresses[rule.Host]
-				envoyIngress.addUpstream(j.Hostname)
+
+				if j.Hostname != "" {
+					envoyIngress.addUpstream(j.Hostname)
+				} else {
+					envoyIngress.addUpstream(j.IP)
+				}
```

Comment:
what happens if neither of these are set?

Reply:
Then we would get the same error as before. But it would also likely mean there is no load balancer to send traffic too.